Explain templates and jsPsych version compatibility (#6)
jspsych/jspsych-contrib: README.md

Both templates are compatible with jsPsych v7+.

Regardless of which template you use, you can get started by creating a copy of the template folder in `/packages` and renaming it according to your plugin/extension name. Be sure that you also:
* Edit the `package.json` file
* Add a readme.md file to your plugin/extension directory, based on the [readme template](readme-template.md)

### `plugin-template-ts`

This template uses TypeScript source files that are complied into JavaScript using Node.js and npm.
This is the format used for plugins in the main jsPsych repo.
To use this template, you should edit the `src/index.ts` file, keeping the overall structure but changing the details as appropriate (plugin name, parameters, trial method, etc.).
You can then use the `npm run build` command to compile your `index.ts` code into JavaScript files, which will appear in a `/dist` directory.
This format also allows you to add a Jest test file (optional), which you can create based on the `src/index.spec.ts` template file.

In the `rollup.config.mjs` file, replace "jsPsychPluginName" with your plugin name (same as the class name in the `index.ts` file).

> You do not need to edit the other config files in this template directory.

Author: Is this right?

Reviewer: I find "same as the class name in the `index.ts` file" a bit confusing, since `jsPsychPluginName` != `PluginName`...

Author: Oh right, oops. I'll remove that. Is it correct to say this?
"In the `rollup.config.mjs` file, replace "jsPsychPluginName" with your plugin name. This will be the name that can be used as the value of the `type` parameter when creating a trial with your plugin."

Author: Done in f1478ee

Reviewer: > Is it correct to say this?

Sorry, didn't see this! Absolutely right!

This could be accomplished with a link to a demonstration experiment, the inclusion of an example file and/or testing files, or through some other means.
We try to review pull requests quickly and add new contributions as soon as the minimal standards are met.

## Plugin templates

> There are two plugin template directories inside the `/packages` directory that you can as a reference when creating a directory for your plugin contribution. Both templates are compatible with jsPsych v7+.

Reviewer: ```suggestion
There are two plugin template directories inside the `/packages` directory that you can use as a reference when creating a directory for your plugin contribution. Both templates are compatible with jsPsych v7+.
```

> * `plugin-template-ts`: This template uses TypeScript source files that are complied into JavaScript using Node.js and npm. This is the format used for plugins in the main jsPsych repo. For more details, please see the jsPsych documentation page [Configuring the jsPsych development environment](https://www.jspsych.org/developers/configuration).

Reviewer: ```suggestion
* `plugin-template-ts`: This template uses TypeScript source files that are complied into JavaScript using Node.js and npm.
  This is the format used for plugins in the main jsPsych repo.
  For more details, including setup instructions, please see the jsPsych documentation page [Configuring the jsPsych development environment](https://www.jspsych.org/latest/developers/configuration).
```
@jodeleeuw Is this enough info for the NPM stuff here? I'll add a warning box about the `npm install` location in the config docs anyway.

Reviewer 2: Yeah this works for me!

* `plugin-template`: This template allows you to put your plugin's JavaScript code directly into a JavaScript template file, rather than using TypeScript and Node.js/npm. 

## jsPsych version compatibility

We would like to encourage you to contribute plugins that are compatible with the latest jsPsych version. At the same time, we realize that there may be jsPsych users who have created very useful plugins with jsPsych v6 that they would like to share with the community, but don't have the time/resources to convert into the jsPsych v7+ Node package format. Therefore we welcome plugins that are compatible with v6 as well as v7+. 

If you'd like to contribute a jsPsych v6 plugin, please do the following:
* Use the `plugin-template` directory as a reference
* Delete everything inside of the `index.js` template file and replace it with your v6-compatible plugin code
* Edit the `package.json` file with the information about your plugin

> * Delete the "devDependencies" property and values in the `package.json` files (this is only relevant for plugins that are compatible with jsPsych v7+)

Reviewer: Since `@jspsych/config` contains babel, it should technically be specified as a dev dependency (while it would still work without that in the monorepo because other packages depend on `@jspsych/config` too, so it's already installed there).
```suggestion
* Change the "jspsych" version field in "devDependencies" in `package.json` to "^6.3.1"
```
I'll push the previous jsPsych releases to NPM today so that will work then.

> * Add a `readme.md` file, which must state the jsPsych version that your plugin is compatible with

Reviewer 2: I created a readme template file, but I think it's in the extensions branch/PR. We can link to that here.

Reviewer 2: Oh wait, that PR is merged now so it's available here: https://github.com/jspsych/jspsych-contrib/blob/main/readme-template.md
